frontend/packages/console-shared/src/hooks/post-form-submit-action.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useMemo } from 'react';
+import {
+  isPostFormSubmissionAction,
+  PostFormSubmissionAction,
+  useExtensions,
+} from '@console/plugin-sdk';
+import { K8sResourceCommon } from '@console/internal/module/k8s';
+import { useQueryParams } from './useQueryParams';
+
+export const usePostFormSubmitAction = <R = K8sResourceCommon[]>(
+  key: string = 'action',
+): ((arg: R) => void) => {
+  const params = useQueryParams();
+  const actionParams = params.get(key);
+
+  const [actionTypes, actionPayload] = useMemo(() => {
+    const parsedAction = JSON.parse(actionParams);
```

Comment:
should we have `JSON.parse`  in try catch block or it's always safe?

Reply:
I think it's safe to use it like this as any action that is passed will be stringified.

Reply:
Since the data is coming from a query string, I think it's a good idea to catch the error.

Reply:
fixed!